include/openssl/core_names.h

```diff
@@ -157,14 +157,31 @@ extern "C" {
 #define OSSL_PKEY_PARAM_DSA_PUB_KEY  "pub"
 #define OSSL_PKEY_PARAM_DSA_PRIV_KEY "priv"
 
+/* RSA Keys */
+#define OSSL_PKEY_PARAM_RSA_N        "n"
+#define OSSL_PKEY_PARAM_RSA_E        "e"
+#define OSSL_PKEY_PARAM_RSA_D        "d"
+#define OSSL_PKEY_PARAM_RSA_DMP1     "dmp1"
+#define OSSL_PKEY_PARAM_RSA_DMQ1     "dmq1"
+#define OSSL_PKEY_PARAM_RSA_IQMP     "iqmp"
```

Comment:
I have a different idea for the last three, which supports multi-prime RSA keys as well.  P and Q, DMP1 and DMQ1 and IQMP really have no special standing visavi multi-prime keys, except that they are the original factors.  So regarding primes and CRT parameters, we really just have an array of factors, and an array of exponents, and an array of coefficients, where the amount of factors and exponents must be equal, and the amount of coefficients must be one less.

If we code the interface between libcrypto and providers with that in mind, the coding becomes *much* simpler, and there's no need for special cases.

The only real reason libcrypto uses the old names plus added multi-prime info is historical; that's how the ASN.1 type could be extended without breaking it for old applications.  We don't have to follow that example since we're making something new, i.e. backward compatibility isn't a concern here.

Reply:
Doing this with `OSSL_PARAM` arrays is already covered, because this:

> Normally, the order of the an **OSSL_PARAM** array is not relevant. However, if the _responder_ can handle multiple elements with the same key, those elements must be handled in the order they are in.

Ref: https://github.com/openssl/openssl/blob/master/doc/man3/OSSL_PARAM.pod

Reply:
I've rebased my work on top of yours - so this is no longer relevant.